tuf/signed/sign.go

```diff
@@ -13,54 +13,72 @@ package signed
 
 import (
 	"crypto/rand"
-	"fmt"
 
 	"github.com/Sirupsen/logrus"
+	"github.com/docker/notary/trustmanager"
 	"github.com/docker/notary/tuf/data"
 	"github.com/docker/notary/tuf/utils"
 )
 
-// Sign takes a data.Signed and a key, calculated and adds the signature
-// to the data.Signed
+// Sign takes a data.Signed and keys, calculates and adds at least
+// minSignature signatures using primaryKeys the data.Signed.  It will also
+// clean up any signatures produced by a key not in the primaryKeys or
+// validSigningKeys list.
+// (validSigningKeys can be either a superset of or a completely disjoint set from
+// primaryKeys)
```

Comment:
Should it be noted that `primaryKeys` are still considered valid keys to produce signatures with?  It might be confusing if `primaryKeys` and `validSigningKeys` are disjoint, but both still completely valid